pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -600,6 +600,12 @@ func (r *ReconcileInfinispan) upgradeInfinispan(infinispan *infinispanv1.Infinis
 		infinispan.Spec.Image = nil
 		updateNeeded = true
 	}
+	sc := infinispan.Spec.Service.Container
+	// Remove defined to "" Spec.Service.Container.Storage field for upgrade backward compatibility
+	if sc != nil && sc.Storage != nil && *infinispan.Spec.Service.Container.Storage == "" {
```

Comment:
I'm worried about all the `if` around to deal with backward compatibilities. 
Can a `webhook` approach be of help here?

Reply:
I'm not sure that will be problem, as we are moving from mandatory to optional field type. I'm waiting for upgrade PR's merge for fully validation

Reply:
I was thinking to webhook as a general way for updating cr during upgrade, instead of adding ad hoc code in the cotroller

Reply:
Yeah, @rigazilla agree with you. But it would be nice to wait for update to the Operator SDK 1.x where webhook implementation much more easy than right now. We can use this impl as is and change it in the future.

Reply:
sure @dmvolod, this PR is ok.
I'll look into SDK 1.x, thanks!